.github/workflows/main.yml

```diff
@@ -16,32 +16,47 @@ jobs:
         runs-on: ubuntu-latest
 
         steps:
-            - uses: actions/checkout@v4
-            - uses: actions/setup-node@v4
+            - uses: actions/checkout@v6
+
+            - name: Install pnpm
+              uses: pnpm/action-setup@v4
+              id: pnpm-install
+              with:
+                  run_install: false
+            - uses: actions/setup-node@v6
               with:
                   node-version: 24
-                  cache: npm
+                  cache: pnpm
+                  cache-dependency-path: pnpm-lock.yaml
 
-            - run: npm ci
-            - run: npm run check
-            - run: npm run build
+            - run: pnpm install
+            - run: pnpm run check:all
+            - run: pnpm run build:all
 
     test:
         runs-on: ubuntu-latest
         strategy:
             fail-fast: false
             matrix:
-                node-version: [18, 24]
+                node-version: [22, 24]
```

Comment:
I know we want to drop Node 18 support, but does this implicitly drop 20 as well? If so, is there reason to do that?